packages/turbo/install.js

```diff
@@ -195,30 +196,35 @@ function maybeOptimizePackage(binPath) {
   // This optimization also doesn't apply when npm's "--ignore-scripts" flag is
   // used since in that case this install script will not be run.
   if (os.platform() !== "win32" && !isYarn()) {
-    const tempPath = path.join(__dirname, "bin-turbo");
-    try {
-      // First link the binary with a temporary file. If this fails and throws an
-      // error, then we'll just end up doing nothing. This uses a hard link to
-      // avoid taking up additional space on the file system.
-      fs.linkSync(binPath, tempPath);
+    const optimizeBin = (from, to, temp) => {
+      const tempPath = path.join(__dirname, temp);
+      try {
+        // First link the binary with a temporary file. If this fails and throws an
+        // error, then we'll just end up doing nothing. This uses a hard link to
+        // avoid taking up additional space on the file system.
+        fs.linkSync(from, tempPath);
 
-      // Then use rename to atomically replace the target file with the temporary
-      // file. If this fails and throws an error, then we'll just end up leaving
-      // the temporary file there, which is harmless.
-      fs.renameSync(tempPath, toPath);
+        // Then use rename to atomically replace the target file with the temporary
+        // file. If this fails and throws an error, then we'll just end up leaving
+        // the temporary file there, which is harmless.
+        fs.renameSync(tempPath, to);
 
-      // If we get here, then we know that the target location is now a binary
-      // executable instead of a JavaScript file.
-      isToPathJS = false;
+        // If we get here, then we know that the target location is now a binary
+        // executable instead of a JavaScript file.
+        isToPathJS = false;
 
-      // If this install script is being re-run, then "renameSync" will fail
-      // since the underlying inode is the same (it just returns without doing
-      // anything, and without throwing an error). In that case we should remove
-      // the file manually.
-      fs.unlinkSync(tempPath);
-    } catch {
-      // Ignore errors here since this optimization is optional
-    }
+        // If this install script is being re-run, then "renameSync" will fail
+        // since the underlying inode is the same (it just returns without doing
+        // anything, and without throwing an error). In that case we should remove
+        // the file manually.
+        fs.unlinkSync(tempPath);
+      } catch {
+        // Ignore errors here since this optimization is optional
```

Comment:
I realize it's probably quite the edge case, but do we need to worry about only one of the binaries successfully copying?

Reply:
Yeah, actually, I'm going to reverse the order, since if we fail to copy `turbo`, I think we're still ok.